### ISSUE TITLE
NullPointer Exception reading bytes

### ISSUE THREAD

**Issue by jarpz** [eclipse-paho/paho.mqtt.java#197]
@jpwsutton ....

Sometimes I'm receiving a null pointer exception when in low level the lib try to read a byte array in line 65 of [MqttInputStream.java](https://github.com/eclipse/paho.mqtt.java/blob/master/org.eclipse.paho.client.mqttv3/src/main/java/org/eclipse/paho/client/mqttv3/internal/wire/MqttInputStream.java) code:

**_byte first = in.readByte();**_

```
public MqttWireMessage readMqttWireMessage() throws IOException, MqttException {
        final String methodName ="readMqttWireMessage";
        ByteArrayOutputStream bais = new ByteArrayOutputStream();
        byte first = in.readByte();
        clientState.notifyReceivedBytes(1);

        byte type = (byte) ((first >>> 4) & 0x0F);
        if ((type < MqttWireMessage.MESSAGE_TYPE_CONNECT) ||
            (type > MqttWireMessage.MESSAGE_TYPE_DISCONNECT)) {
            // Invalid MQTT message type...
            throw ExceptionHelper.createMqttException(MqttException.REASON_CODE_INVALID_MESSAGE);
        }
        long remLen = MqttWireMessage.readMBI(in).getValue();
        bais.write(first);
        // bit silly, we decode it then encode it
        bais.write(MqttWireMessage.encodeMBI(remLen));
        byte[] packet = new byte[(int)(bais.size()+remLen)];
        readFully(packet,bais.size(),packet.length - bais.size());

        byte[] header = bais.toByteArray();
        System.arraycopy(header,0,packet,0, header.length);
        MqttWireMessage message = MqttWireMessage.createWireMessage(packet);
        // @TRACE 501= received {0} 
        log.fine(CLASS_NAME, methodName, "501",new Object[] {message});
        return message;
    }
```

Maybe, the client cannot read the message from the broker

This is the [crashlytics](http://crashes.to/s/900625b5f57) log, I'm using a ssl connection with mosquitto broker...

**Comment by jpwsutton** [comment]
Hi, is this using the released version or the 1.0.3-SNAPSHOT version?

**Comment by vanshg** [comment]
I'm getting this same error. I think it might have something to do with an improper SSL connection

**Comment by jpwsutton** [comment]
Could you provide any steps to reproduce this error please?

**Comment by vanshg** [comment]
Yeah sure. I began using the plain Java client because of some other bugs in the Android client that I was encountering in 7.0 Nougat. Originally with the Android client (which I believe uses this client in its implementation) worked perfectly fine. However, once I started using the standalone AsyncClient I get this error when connecting to an SSL mqtt broker. However the connection works fine for other brokers and when connecting to the same broker via the Android client on older OSes. The exception message I get is "Connection lost"
